purchase_sale_stock_inter_company/tests/test_inter_company_purchase_sale_stock.py

```diff
@@ -0,0 +1,396 @@
+                ]
+            }
+        )
+        sale = self._approve_po()
+        sale_picking = sale.picking_ids
+        self.assertEqual(len(sale.picking_ids), 1)
+        sale_picking.sudo().action_confirm()
+        for move in sale_picking.move_ids:
+            move.quantity = move.product_uom_qty
+        sale_picking.sudo().button_validate()
+        self.assertEqual(len(self.purchase_company_a.picking_ids), 1)
+        self.assertEqual(len(self.purchase_company_a.picking_ids.move_line_ids), 2)
+
+    def test_sync_picking_lot_and_qty_with_move_diff(self):
+        """
+        Create a purchase order with 3 lines for 3 different products:
+        - stockable product tracked by serial (1): 1 qty
+        - stockable product tracked by serial (2): 3 qty
+        - stockable product (not tracked by serial): 50 qty
+        In the delivery, having this:
+        - stockable product tracked by serial (1): 2 move lines with 1 lot each
+        - stockable product tracked by serial (2): 1 move line with 1 lot
+        - stockable product (not tracked by serial): 1 move line with 30 qty and no lot
+        Expected to have the same move lines in the receipt
+        """
+        po_form = Form(self.env["purchase.order"])
+        po_form.company_id = self.company_a
+        po_form.partner_id = self.partner_company_b
+
+        with po_form.order_line.new() as line_form:
+            line_form.product_id = self.stockable_product_serial
+            line_form.product_qty = 1
+            line_form.price_unit = 450.0
+        with po_form.order_line.new() as line_form:
+            line_form.product_id = self.stockable_product_serial_2
+            line_form.product_qty = 3
+            line_form.price_unit = 450.0
+        with po_form.order_line.new() as line_form:
+            line_form.product_id = self.stockable_product
+            line_form.product_qty = 50
+            line_form.price_unit = 1000.0
+        self.purchase_company_a = po_form.save()
+
+        purchase = self.purchase_company_a
+        sale = self._approve_po()
+
+        # validate the SO picking
+        po_picking_id = purchase.picking_ids
+        so_picking_id = sale.picking_ids
+        so_moves = so_picking_id.move_ids
+        for move in so_moves:
+            if move.product_id == self.stockable_product_serial:
+                move.move_line_ids = [
+                    (
+                        0,
+                        0,
+                        {
+                            "location_id": move.location_id.id,
+                            "location_dest_id": move.location_dest_id.id,
+                            "product_id": self.stockable_product_serial.id,
+                            "quantity": 1,
+                            "lot_id": self.serial_1.id,
+                            "picking_id": so_picking_id.id,
+                        },
+                    ),
```

Comment:
Please use [Command.create](https://github.com/odoo/odoo/blob/9748b45b24140416fc6e0e87c25c60285bef3d70/odoo/fields.py#L4259); it's more readable.

```suggestion
                    Command.create(
                        {
                            "location_id": move.location_id.id,
                            "location_dest_id": move.location_dest_id.id,
                            "product_id": self.stockable_product_serial.id,
                            "quantity": 1,
                            "lot_id": self.serial_1.id,
                            "picking_id": so_picking_id.id,
                        },
                    ),
```